src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1114,72 +1114,83 @@ static MemorySegment ofAddress(long address, long byteSize, MemorySession sessio
     }
 
     /**
-     * Creates a native segment with the given layout.
+     * Creates a native segment with the given layout and memory session.
+     * <p>
+     * Clients are responsible for ensuring that the memory session associated with the returned segment is
+     * closed when segments are no longer in use. Failure to do so will result in off-heap memory leaks. As an
+     * alternative to explicitly invoke {@link MemorySession#close()}, sessions backed with a Cleaner
+     * (such as {@linkplain MemorySession#openImplicit()}) can be used, allowing the returned segment to be
+     * automatically released some unspecified time after the session is no longer referenced.
      * <p>
      * The {@linkplain #address() address} of the returned memory segment is the starting address of
      * the newly allocated off-heap region backing the segment. Moreover, the {@linkplain #address() address}
      * of the returned segment will be aligned according to the alignment constraint of the provided layout.
      * <p>
-     * The returned memory segment is associated with a new {@linkplain MemorySession#openImplicit implicit}
-     * memory session. As such, the off-heap region which backs the returned segment is
-     * freed <em>automatically</em>, some unspecified time after it is no longer referenced.
-     * <p>
-     * Native segments featuring deterministic deallocation can be obtained using the
-     * {@link MemorySession#allocate(MemoryLayout)} method.
-     * <p>
      * This is equivalent to the following code:
      * {@snippet lang=java :
-     * MemorySession.openImplicit()
-     *         .allocate(layout.bytesSize(), layout.bytesAlignment());
+     * session.allocate(layout.bytesSize(), layout.bytesAlignment());
      * }
      * <p>
      * The region of off-heap region backing the returned native segment is initialized to zero.
      *
      * @param layout the layout of the off-heap memory region backing the native segment.
+     * @param session the session to which the returned segment is associated.
      * @return a new native segment.
+     * @throws IllegalStateException if the {@code session} is not {@linkplain MemorySession#isAlive() alive}.
```

Comment:
These two throws clause seem off. In other places we use:

```
* @throws IllegalStateException if the {@linkplain #session() session} associated with this segment is not
* {@linkplain MemorySession#isAlive() alive}.
* @throws WrongThreadException if this method is called from a thread other than the thread owning
* the {@linkplain #session() session} associated with this segment.
```

Reply:
We can't use the above "as is" because it says "this segment" - but I think we could use:
```
* @throws IllegalStateException if the {@code session} is not {@linkplain MemorySession#isAlive() alive}.
* @throws WrongThreadException if this method is called from a thread other than the thread owning
* {@code session}.
```